### PR TITLE
CarManagement Service

### PR DESCRIPTION
**Pull request by KiSSel3**

### CODE REVIEW

File: src/Services/CarManagementService/CarManagementService.Presentation/Controllers/CarController.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        {
+            result.TotalCount,
+            result.PageSize,
+            result.CurrentPage,
+            result.TotalPages,
+            result.HasNext,
+            result.HasPrevious
+        };
+
+        Response.Headers.Append("X-Pagination", JsonConvert.SerializeObject(metadata));
```
Comment (TarasovaVeronika) [inline src/Services/CarManagementService/CarManagementService.Presentation/Controllers/CarController.cs]: What do you think about custom page list (smth like as new response model)?  Is it better than your realization?

File: src/Services/CarManagementService/CarManagementService.Presentation/Controllers/CarController.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    private readonly IMapper _mapper;
+
+    public CarController(IMediator mediator, IMapper mapper)
+    {
+        _mediator = mediator;
+        _mapper = mapper;
+    }
+    
+    [HttpPost("get-by-parameters")]
+    public async Task<IActionResult> GetCarsByParametersAsync([FromForm] CarParametersRequestDTO request, CancellationToken cancellationToken = default)
```
Comment (TarasovaVeronika) [inline src/Services/CarManagementService/CarManagementService.Presentation/Controllers/CarController.cs]: Can we use [FromBody] for request? I think that FromForm better when we want to work with sending files for example

File: src/Services/CarManagementService/CarManagementService.Application/UseCases/Commands/Brand/CreateBrand/CreateBrandCommand.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+using MediatR;
+
+namespace CarManagementService.Application.UseCases.Commands.Brand.CreateBrand;
```
Comment (TarasovaVeronika) [inline src/Services/CarManagementService/CarManagementService.Application/UseCases/Commands/Brand/CreateBrand/CreateBrandCommand.cs]: What do you think about change you application disign? 
Like as: UseCase -> Command, Query, CommandHandler, QueryHandlers -> Command/Qery/Handlers looks like (Brand, Car and ect folders)

File: src/Services/CarManagementService/CarManagementService.Application/UseCases/Commands/Brand/CreateBrand/CreateBrandCommandValidator.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+using FluentValidation;
+
+namespace CarManagementService.Application.UseCases.Commands.Brand.CreateBrand;
+
+public class CreateBrandCommandValidator : AbstractValidator<CreateBrandCommand>
+{
+    public CreateBrandCommandValidator()
+    {
+        RuleFor(command => command.Name)
```
Comment (TarasovaVeronika) [inline src/Services/CarManagementService/CarManagementService.Application/UseCases/Commands/Brand/CreateBrand/CreateBrandCommandValidator.cs]: May be can be better to validate request model in API layer?

File: src/Services/CarManagementService/CarManagementService.Application/UseCases/Queries/CarModel/GetCarModelByBrandIdAndName/GetCarModelByBrandIdAndNameQueryHandler.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public async Task<CarModelDTO> Handle(GetCarModelByBrandIdAndNameQuery request, CancellationToken cancellationToken)
+    {
+        _logger.LogInformation("Fetching car model for Brand ID: {BrandId} and Name: {Name}", request.BrandId, request.Name);
+
+        var carModel = await _repository.GetByBrandIdAndNameAsync(request.BrandId, request.Name, cancellationToken);
+        if (carModel is null)
+        {
+            throw new EntityNotFoundException($"Car model with Brand ID {request.BrandId} and Name {request.Name} was not found.");
```
Comment (TarasovaVeronika) [inline src/Services/CarManagementService/CarManagementService.Application/UseCases/Queries/CarModel/GetCarModelByBrandIdAndName/GetCarModelByBrandIdAndNameQueryHandler.cs]: What do you think about adding errors' loggin?

**Review by TarasovaVeronika — CHANGES_REQUESTED** [review]

File: src/Services/CarManagementService/CarManagementService.Application/UseCases/Queries/CarModel/GetCarModelByBrandIdAndName/GetCarModelByBrandIdAndNameQueryHandler.cs (hunk 5 above)
Comment (KiSSel3) [reply src/Services/CarManagementService/CarManagementService.Application/UseCases/Queries/CarModel/GetCarModelByBrandIdAndName/GetCarModelByBrandIdAndNameQueryHandler.cs]: Errors are logged at the global exception handler level

**Review by TarasovaVeronika — APPROVED** [review]

**Pull request merged**